frontend/public/components/storage/create-pvc.tsx

```diff
@@ -0,0 +1,368 @@
+            required
+          />
+        </div>
+        <p className="form-text text-muted" id="pvc-name-help">
+          A unique name for the storage claim within the project.
+        </p>
+        <label className="control-label co-required" htmlFor="access-mode">
+          Access Mode
+        </label>
+        <div className="form-group">
+          {this.state.radios.map(radio => {
+            const checked = radio.value === this.state.accessMode;
+            return (
+              <RadioInput
+                {...radio}
+                key={radio.value}
+                onChange={this.handleChange}
+                checked={checked}
+                aria-describedby="access-mode-help"
+                className="radio-inline"
+                name="accessMode"
+              />
+            );
+          })}
+          <p className="form-text text-muted" id="access-mode-help">
+            Permissions to the mounted drive.
+          </p>
+        </div>
+        <label className="control-label co-required" htmlFor="request-size-input">
+          Size
+        </label>
+        <RequestSizeInput
+          name="requestSize"
+          required={false}
+          onChange={this.handleChange}
+          placeholder={requestSizeValue}
+          defaultRequestSizeUnit={requestSizeUnit}
+          dropdownUnits={dropdownUnits}
+          ariaName="requestSize-helpID"
+        />
+        <p className="form-text text-muted" id="requestSize-help">
+          Desired storage capacity.
+        </p>
+        <Checkbox
+          label="Use label selectors to request storage"
+          onChange={this.toggleShowLabelSelectors}
+          checked={showLabelSelectors}
+          name="showLabelSelector"
+        />
+        <div className="form-group">
+          {showLabelSelectors && (
+            <NameValueEditor
+              nameValuePairs={nameValuePairs}
+              valueString="Selector"
+              nameString="Label"
+              addString="Add Value"
+              readOnly={false}
+              allowSorting={false}
+              updateParentData={this.handleNameValuePairs}
+            />
+          )}
+          <p className="form-text text-muted" id="label-selector-help">
+            Use label selectors to define how storage is created.
+          </p>
+        </div>
+      </div>
+
+    );
+  }
+}
+
+class CreatePvcPage extends React.Component<CreatePvcPageProps, CreatePvcPageState> {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      error: '',
+      inProgress: false,
+      title: 'Create Storage',
+      pvcObj: {},
+    };
+  }
+
+  onChange = pvcObj => {
+    this.setState({ pvcObj });
+  };
+
+  validateForm = () => {
+    if (!_.get(this.state.pvcObj, 'metadata.name')) {
+      this.setState({ error: 'Persistent volume claim name must be set.' });
+      return false;
+    }
+    return true;
+  };
+
+  save = e => {
+    if (!this.validateForm()) {
```

Comment:
This check works well for testing that might have headless usage. Also,  I like to have control over the error messages for things that could go wrong instead of relying on the API's sometime cryptic messages.

Reply:
But there is no point of having control over the error message since the real validation is done by the html *required* attribute.
You will need to have a more complex name validation to make sense of it, cause now its only checking if there is anything in the input field. All the other validation is done by API which we should rely on.

Reply:
I do understand your point, especially since I do the regex on the form side to validate the name.  I am fine removing this. @spadgett do you have any concerns with removing the save validation of the form?

Reply:
I think it would simplify things removing the checks and using HTML5 form validation since it's consistent with other forms. The one place that won't work is with the `Dropdown` component since it's not a native input, although you should get a reasonable error on save from the API server.

We might take a step back and look at how we do validation across forms in the console, but that's outside the scope of this PR. Let's try to be consistent where we can in the meantime.